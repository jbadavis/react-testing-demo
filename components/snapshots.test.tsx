import {render} from '@testing-library/react';
import Snapshots from './snapshots';

describe('Snapshots', () => {
  test('renders correctly', () => {
    const {container} = render(<Snapshots />);

    expect(container).toMatchSnapshot();
  });
});
