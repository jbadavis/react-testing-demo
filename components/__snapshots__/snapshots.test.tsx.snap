// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders correctly 1`] = `
<div>
  <h2
    class="chakra-heading css-jmbq56"
  >
    Snapshot Testing
  </h2>
  <div
    class="chakra-stack css-1sam6cv"
  >
    <p
      class="chakra-text css-0"
    >
      Snapshot tests are a great way of ensuring that any changes don't unexpectedly change the output of other components and affect the wider UI.
    </p>
    <p
      class="chakra-text css-0"
    >
      A typical snapshot tests renders a UI component, takes a snapshot, then creates or compares it to a reference snapshot file stored alongside the test. A snapshot test will fail if there is any mismatch between the new and reference snapshot. Either a change needs to be reverted or the snapshot needs to be updated.
    </p>
  </div>
</div>
`;
